text/0080-datetime-extension.md

```diff
@@ -0,0 +1,367 @@
+- `DT1 == DT2` returns `true` when `DT1` is equal to `DT2`
+
+Values of type `datetime` can also be used with the `-` operator:
+
+- `DT1 - DT2` returns the difference between `DT1` and `DT2` as a `duration`.
+
+Note that the inverse of `-` is `.add(duration)`, not `+`.
+
+The `datetime` type is internally represented as a `long` and contains a Unix Time in milliseconds. This is the number of non-leap seconds that have passed since `1970-01-01T00:00:00Z` in milliseconds. Unix Time days are always 86,400 seconds and handle leap seconds by absorbing them at the start of the day. Due to using Unix Time, and not providing a "current time" function, Cedar avoids the complexities of leap second handling, pushing them to the system and application.
+
+- `duration(long, string)` constructs a duration value. The string argument must be one of `"days", "hours", "minutes", "seconds", "milliseconds"`. Strict validation requires `string` to be a string literal, although evaluation/authorization support any string-typed expression.
+- `.toMilliseconds()` returns a `long` describing the number of milliseconds in this duration. (the value as a long, itself)
+- `.toSeconds()` returns a `long` describing the number of seconds in this duration. (`.toMilliseconds() / 1000`)
+- `.toMinutes()` returns a `long` describing the number of minutes in this duration. (`.toSeconds() / 60`)
+- `.toHours()` returns a `long` describing the number of hours in this duration. (`.toMinutes() / 60`)
+- `.toDays()` returns a `long` describing the number of days in this duration. (`.toHours() / 24`)
+
+Values with type `duration` duration can also be used with comparison operators:
+
+- `DT1 < DT2` returns `true` when `DT1` is before `DT2`
+- `DT1 <= DT2` returns `true` when `DT1` is before or equal to `DT2`
+- `DT1 > DT2` returns `true` when `DT1` is after `DT2`
+- `DT1 >= DT2` returns `true` when `DT1` is after or equal to `DT2`
+- `DT1 == DT2` returns `true` when `DT1` is equal to `DT2`
+
+### Out of scope
+
+- **Conversion between UTC and epochs:** This will be particularly difficult to model and verify in Lean (although it's technically possible, see [this paper](https://dl.acm.org/doi/abs/10.1145/3636501.3636958) which does something similar in the Coq proof assistant). Since it will likely require input-dependent loops, it is unlikely that this can be reasoned about efficiently with SMT.
+
+- **Conversion between UTC and other Time Zones:** Time Zones are a notoriously complex system that evolves rapidly. We avoid this complexity by offering `datetime.add(duration).` Policy authors that require "local time" can either provide an additional datetime in `context` or provide a `duration` to the `context` and call `.add()` to shift the time.
+
+- **Leap Seconds:** Cedar does not have a clock, and this proposal does not add one. Instead, we pass Unix Time through `context` or an entity and let the system / application handle leap seconds.
+
```

Comment:
Reword "Instead, we pass Unix Time through `context` or an entity" to "Instead, the application passes the current time through `context` or an entity attribute". 

If you want to say that a DateTime is equivalent in its representation to Unix Time you should say that somewhere else. Indeed, I think you should give the underlying representation here because it needs to be known for the entity JSON format.

Reply:
In general, I'd like to see more technical details.

- How will you implement policy evaluation for these operators? Depends on their representation.
- How will we implement validation? Seems we need to change the way existing operators are treated. What's the effect there?
- What is the strategy for SMT/analysis?

Reply:
We've added more details and better signposting in the latest revision, which will hopefully answer these questions 🙂 Would appreciate a re-review when you get a chance!